conan/tools/cmake/cmakedeps.py

```diff
@@ -626,6 +626,13 @@ def generate(self):
             generator_file = os.path.join(self.output_path, generator_file)
             save(generator_file, content)
 
+    def _data_filename(self, pkg_filename):
+        data_fname = "{}-{}".format(pkg_filename, self.configuration.lower())
+        if self.arch:
```

Comment:
This is just a detail: If we are already using build_type `none` when the build_type does not exist, why not doing the same with the architecture and follow the same filename pattern?

Reply:
That's a good observation. I didn't realize that `none` was a valid pattern. @memsharded what do you prefer?

Reply:
I just realized: If the build type is not declared and you do: `str(self._conanfile.settings.build_type)` it will crash, right? I think the `build_type` is right now mandatory in this generator as it was in the `cmake_find_package_multi`

Reply:
build_type ``none`` might be undefined behavior, I don't think the generator expressions in the targets etc will handle this correctly. We need to figure out the multiplicity (build_type, arch) of the generator, and how they connect the producer build_type and the consumer build_type.


So I suggest, leave it as-is, whatever it passes current tests is already a big improvement, and we will address this later.